toffy/watcher_callbacks.py

```diff
@@ -1,81 +1,157 @@
-            Additional arguments for `mibi_bin_tools.bin_files.extract_bin_files`.
-            Accepted kwargs are:
-
-         - intensities
-         - time_res
-
-    Returns:
-        Callable[[str, str, str], None]:
-            Callback for fov watcher
-    """
-
-    if isinstance(panel, tuple):
-        raise TypeError('Global unit mass integration is no longer support. Please provide panel '
-                        'as a pandas DataFrame...')
-
-    if not os.path.exists(out_dir):
-        os.makedirs(out_dir)
-
-    def extract_callback(run_folder: str, point_name: str):
-        if not os.path.exists(out_dir):
-            os.makedirs(out_dir)
-
-        extract_bin_files(run_folder, out_dir, [point_name], panel, **kwargs)
-
-    return extract_callback
+        for metric_name, data in metric_data.items():
+            data.to_csv(os.path.join(qc_out_dir, metric_name), index=False)
 
 
-def build_qc_callback(out_dir: str, panel: pd.DataFrame, **kwargs) -> Callable[[str, str], None]:
-    """Generates qc callback for given panel + parameters
+def build_fov_callback(*args, **kwargs):
+    """Assembles callbacks to be run for each transfered FoV
 
     Args:
-        out_dir (str):
-            Path where qc metrics are written
-        panel (pd.DataFrame):
-            Target mass integration ranges
-        **kwargs (dict):
-            Additional arguments for `toffy.qc_comp.compute_qc_metrics`.  Accepted kwargs are:
-
-         - gaussian_blur
-         - blur_factor
-
-    Returns:
-        Callable[[str, str], None]:
-            Callback for fov watcher
+        *args (List[str]):
+            Names of member functions of `FovCallbacks` to chain together
+        **kwargs (Dict[str, Any]):
+            Arguments to pass to `FovCallbacks` member functions specified in *args
+
+    Raises:
+        ValueError:
+            Raised on non-existant member function or missing required kwarg
     """
-
-    if isinstance(panel, tuple):
-        raise TypeError('Global unit mass integration is no longer support. Please provide panel '
-                        'as a pandas DataFrame...')
-
-    kwargs['save_csv'] = False
-
-    if not os.path.exists(out_dir):
-        os.makedirs(out_dir)
-
-    def qc_callback(run_folder: str, point_name: str):
-        metric_data = compute_qc_metrics(run_folder, point_name, None, panel, **kwargs)
-        for metric_name, data in metric_data.items():
-            data.to_csv(os.path.join(out_dir, metric_name), index=False)
-
-    return qc_callback
+    # retrieve all 'non-special' methods of FovCallbacks
+    methods = [attr for attr in dir(FovCallbacks) if attr[0] != '_']
+
+    # validate user callback settings
+    for arg in args:
+        # check that `arg` is a method of FovCallbacks
+        if arg not in methods:
+            raise ValueError(
+                f'{arg} is not a valid FovCallbacks member\n'
+                f'Accepted callbacks are {methods}'
+            )
```

Comment:
Would it be better to use something like `misc_utils.verify_in_list` to print a list of invalid `args` in the case of multiple, or do you think it's fine to just list the first offending one?

Reply:
good idea, `verify_in_list` is probably better here.